WordPress/Classes/ViewRelated/Me/My Profile/Change Username/ChangeUsernameViewController.swift

```diff
@@ -135,44 +145,50 @@ private extension ChangeUsernameViewController {
                                                 preferredStyle: .alert)
         alertController.addAttributeMessage(String(format: Constants.Alert.message, viewModel.selectedUsername),
                                             highlighted: viewModel.selectedUsername)
-        alertController.addCancelActionWithTitle(Constants.Alert.cancel, handler: { [weak alertController] _ in
-            if let textField = alertController?.textFields?.first {
-                NotificationCenter.default.removeObserver(textField, name: UITextField.textDidChangeNotification, object: nil)
-            }
+        alertController.addCancelActionWithTitle(Constants.Alert.cancel, handler: { _ in
             DDLogInfo("User cancelled alert")
         })
-        changeUsernameAction = alertController.addDefaultActionWithTitle(Constants.Alert.change, handler: { [weak alertController] _ in
-            guard let textField = alertController?.textFields?.first,
+        let action = alertController.addDefaultActionWithTitle(Constants.Alert.change, handler: { [weak alertController, weak self] _ in
+            guard let self, let alertController else { return }
+            guard let textField = alertController.textFields?.first,
                 textField.text == self.viewModel.selectedUsername else {
                     DDLogInfo("Username confirmation failed")
                     return
             }
             DDLogInfo("User changes username")
-            NotificationCenter.default.removeObserver(textField, name: UITextField.textDidChangeNotification, object: nil)
```

Comment:
This is the same unregistering observer issue as [the above comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/21003/files#r1251461737).